packages/mongo-nav/src/mongo-select/MongoSelect.tsx

```diff
@@ -89,140 +104,137 @@ const onKeyDown: React.KeyboardEventHandler = e => {
-    props: ProjectMongoSelectProps,
-    datum: ProjectInterface,
-  ) => {
-    const content = (
-      <ProjectOption
-        projectName={datum.projectName}
-        href={props.constructProjectURL(datum.orgId, datum.projectId)}
-      />
-    );
+function OrgSelect({
+  current,
+  data,
+  className,
+  urls,
+  isActive,
+  onChange,
+  onClick,
+  constructOrganizationURL,
+}: OrganizationMongoSelectProps) {
+  const renderOrganizationOption = (datum: OrganizationInterface) => {
+    const { orgId, orgName, planType } = datum;
 
     return (
       <MenuItem
-        key={datum.projectId}
+        key={orgId}
         className={menuItemContainerStyle}
         onClick={onClick}
+        href={constructOrganizationURL(orgId)}
       >
-        {content}
+        <div className={orgOptionContainerStyle}>
+          <span className={nameStyle}>{orgName}</span>
+          <span className={productStyle}>{formatPlanType(planType)}</span>
+        </div>
       </MenuItem>
     );
   };
 
-  const renderOrganizationOption = (
-    props: OrganizationMongoSelectProps,
-    datum: OrganizationInterface,
-  ) => {
-    const content = (
-      <OrganizationOption
-        orgName={datum.orgName}
-        planType={datum.planType}
-        href={props.constructOrganizationURL(datum.orgId)}
-      />
-    );
+  const errorMessage =
+    'You do not belong to any organizations. Create an organization to start using MongoDB Cloud';
+
+  return (
+    <Menu
+      trigger={
+        <OrganizationTrigger
+          current={current?.orgName ?? 'All Organizations'}
+          className={className}
+          urls={urls}
+          isActive={isActive}
+        />
+      }
+      className={menuContainerStyle}
+      justify="start"
+    >
+      <FocusableMenuItem>
+        <Input
+          onChange={onChange}
+          onKeyDown={onKeyDown}
+          variant="organization"
+        />
+      </FocusableMenuItem>
+
+      <>
+        {data ? (
+          data.map(datum => renderOrganizationOption(datum))
+        ) : (
+          <li>{errorMessage}</li>
+        )}
+      </>
+
+      <MenuSeparator />
+      <MenuItem
+        onKeyDown={onKeyDown}
+        href={urls.mongoSelect?.viewAllOrganizations}
+      >
+        <strong className={viewAllStyle}>View All Organizations</strong>
+      </MenuItem>
+    </Menu>
+  );
+}
+
+OrgSelect.displayName = 'OrgSelect';
+
+export { OrgSelect };
+
+function ProjectSelect({
+  current,
+  className,
+  onChange,
+  data,
+  onClick,
+  constructProjectURL,
+  urls,
+}: ProjectMongoSelectProps) {
+  const renderProjectOption = (datum: ProjectInterface) => {
+    const { projectId, projectName, orgId } = datum;
 
     return (
       <MenuItem
-        key={datum.orgId}
+        key={projectId}
         className={menuItemContainerStyle}
         onClick={onClick}
+        href={constructProjectURL(orgId, projectId)}
       >
-        {content}
+        <span className={nameStyle}>{projectName}</span>
```

Comment:
Is there a reason we don't apply the `nameStyle` `className` directly to `MenuItem` rather than rendering a nested span?

Reply:
👍